dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -496,6 +496,14 @@ def get_session_for_workshop_and_day(workshop, day)
       day > 0 ? workshop.sessions[day - 1] : nil
     end
 
+    def get_last_valid_day(workshop)
```

Comment:
Could this just be a method on the workshop model and you reference it like `workshop.last_valid_day`?

Reply:
+1 for factoring this common code, and putting it in the model.  (When in doubt, put business logic in the model.)

Reply:
good point